src/hooks/useCurrenciesFetch.ts

```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { currenciesFetch, selectShouldFetchCurrencies} from '../modules';
+
+export const useCurrenciesFetch = () => {
+    const shouldDispatch = useSelector(selectShouldFetchCurrencies);
+    const dispatch = useDispatch();
+
+    React.useEffect(() => {
+        if (shouldDispatch) {
+            dispatch(currenciesFetch());
```

Comment:
no dispatch in any hook, since hook is called each render an you can got a loop.

Reply:
This is for all hooks in this PR

Reply:
https://react-redux.js.org/api/hooks#recipe-useactions